planning/behavior_velocity_planner/src/utilization/path_utilization.cpp

```diff
@@ -205,44 +207,7 @@ autoware_auto_planning_msgs::msg::Path interpolatePath(
     return path;
   }
 
-  // Interpolate
-  const auto x_interp = interpolation::slerp(s_in, x, s_out);
-  const auto y_interp = interpolation::slerp(s_in, y, s_out);
-  const auto z_interp = interpolation::slerp(s_in, z, s_out);
-
-  // Find a nearest segment for each point in s_out and use the velocity of the segment's beginning
-  // point. Note that if s_out is almost the same value as s_in within DOUBLE_EPSILON range, the
-  // velocity of s_out should be same as the one of s_in.
-  std::vector<double> v_interp;
-  size_t closest_segment_idx = 0;
-  for (size_t i = 0; i < s_out.size() - 1; ++i) {
-    for (size_t j = closest_segment_idx; j < s_in.size() - 1; ++j) {
-      if (s_in.at(j) - DOUBLE_EPSILON < s_out.at(i) && s_out.at(i) < s_in.at(j + 1)) {
-        // find closest segment in s_in
-        closest_segment_idx = j;
-      }
-    }
-    v_interp.push_back(v.at(closest_segment_idx));
-  }
-  v_interp.push_back(v.back());
-
-  // Insert path point to interpolated_path
-  for (size_t idx = 0; idx < v_interp.size() - 1; ++idx) {
-    autoware_auto_planning_msgs::msg::PathPoint path_point;
-    path_point.pose.position.x = x_interp.at(idx);
-    path_point.pose.position.y = y_interp.at(idx);
-    path_point.pose.position.z = z_interp.at(idx);
-    path_point.longitudinal_velocity_mps = v_interp.at(idx);
-    const double yaw =
-      std::atan2(y_interp.at(idx + 1) - y_interp.at(idx), x_interp.at(idx + 1) - x_interp.at(idx));
-    tf2::Quaternion quat;
-    quat.setRPY(0, 0, yaw);
-    path_point.pose.orientation = tf2::toMsg(quat);
-    interpolated_path.points.push_back(path_point);
-  }
-  interpolated_path.points.push_back(path.points.back());
-
-  return interpolated_path;
+  return motion_utils::resamplePath(path, s_out);
```

Comment:
@purewater0901 
Does s_out includes the last point like `interpolated_path.points.push_back(path.points.back());` ?

Reply:
https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_velocity_planner/src/utilization/path_utilization.cpp#L173-L182

since s_in includes the last point, s_out also includes the last point as well. So there should be no problems

Reply:
![image](https://user-images.githubusercontent.com/65527974/182294812-5eb6da82-201b-471b-be53-3388d375b473.png)
In my environment this warning is shown so interpolation looks failed
```
[component_container_mt-13] [motion_utils]: input path size, input path length or resampled arclength is wrong
```

Reply:
Okay, I'll check it

Reply:
Thank you

Reply:
I just updated the code, could you try it again?

Reply:
@purewater0901 
Is there any module which insert slow velocity before goal?
![Uploading image.png…]()